File: backend/src/main/java/com/aiexam/service/QuestionService.java
package com.aiexam.service;

import com.aiexam.dto.CreateQuestionRequest;
import com.aiexam.dto.QuestionResponse;
import com.aiexam.dto.UpdateQuestionRequest;
import com.aiexam.enums.DifficultyLevel;
import com.aiexam.enums.QuestionType;

import java.util.List;

public interface QuestionService {

    QuestionResponse createQuestion(CreateQuestionRequest request);

    QuestionResponse updateQuestion(String questionId,
                                    UpdateQuestionRequest request);

    QuestionResponse getQuestionById(String questionId);

    List<QuestionResponse> getQuestionsByExam(String examId);

    List<QuestionResponse> getAllQuestions();

    List<QuestionResponse> getQuestionsByCourse(String courseId);

    List<QuestionResponse> getQuestionsBySubject(String subjectId);

    List<QuestionResponse> getQuestionsByTeacher(String teacherId);

    List<QuestionResponse> getQuestionsByDifficulty(DifficultyLevel difficulty);

    List<QuestionResponse> getQuestionsByType(QuestionType questionType);

    void enableQuestion(String id);

    void disableQuestion(String id);

    void deleteQuestion(String questionId);

}

File: frontend/src/services/questionService.js
import api from "../api/axios";

const questionService = {

    getQuestionsByExam(examId) {
        return api.get(`/questions/exam/${examId}`);
    },

    getQuestionById(id) {
        return api.get(`/questions/${id}`);
    },

    createQuestion(data) {
        return api.post("/questions", data);
    },

    updateQuestion(id, data) {
        return api.put(`/questions/${id}`, data);
    },

    deleteQuestion(id) {
        return api.delete(`/questions/${id}`);
    }

};

export default questionService;